cmd/nerdctl/run.go

```diff
@@ -966,6 +966,35 @@ func parseKVStringsMapFromLogOpt(cmd *cobra.Command, logDriver string) (map[stri
 		if _, ok := logOptMap[logging.MaxSize]; !ok {
 			delete(logOptMap, logging.MaxFile)
 		}
+		for _, value := range logging.FluentdLogOpts {
+			if _, ok := logOptMap[value]; ok {
+				logrus.Warnf("log-opt %s is ignored for json-file log driver", value)
+			}
+		}
```

Comment:
Why not check the other way around?

```suggestion
		for opt := range logOptMap {
			if strutil.InStringSlice(logging.JSONDriverLogOpts, opt) {
			        logrus.Warnf("log-opt %q is ignored for json-file log driver", opt)
			}
		}
```

Reply:
The hasmap is a O(1) lookup action(

It's just personal style(

Reply:
We can't change it every time we add a new driver